Adds endpoint for client.queryStorageDeal

hazmatzo opened the pull request:
Adds endpoint to query a storage deal

Solves issue [#16 
](https://github.com/filecoin-project/js-filecoin-api-client/issues/16)

**Endpoint**
`/api/client/query-storage-deal`

**Description**
Query a storage deal's status

**Arguments**
* arg [string]: CID of deal to query Required: yes.

**Response**
On success, the call to this endpoint will return with 200 and the following body:

```
{
  "State":7,
  "Message":"",
  "ProposalCid":
    {
      "/":"zDPWYqFD8CNXu7Mo9qPSUANbTK2vi9vJBnvavF9S1pVGPHafVHpT"
    },
  "ProofInfo":
    {
      "SectorID":1,
      "CommitmentMessage":
        {
          "/":"zDPWYqFCtHkWNkE2p6t6TeV1sPP5kbnKc5ajUhMVV8xvrw1u5F1R"
        },
      "PieceInclusionProof":"EiAbbOy4pChsCYqFYA6qJaUJYStlnwYMdQPHZX7YBkVXDD6vgmGTPnWrcdA9M0oAXQCzOq735YKySLUoTI6pAw=="
    },
  "Signature":"c2lnbmF0dXJycmVlZQ=="
}
```

resolves #16

hazmatzo commented:
@alanshaw

alanshaw commented on API.md:
```diff
@@ -401,6 +401,55 @@ After first iteration:
+      "PieceInclusionProof":"EiAbbOy4pChsCYqFYA6qJaUJYStlnwYMdQPHZX7YBkVXDD6vgmGTPnWrcdA9M0oAXQCzOq735YKySLUoTI6pAw=="
+    },
+  "Signature":"c2lnbmF0dXJycmVlZQ=="
+}
```
To make this more idiomatic for JS users we should convert the root and `ProofInfo` to camel case using the [`toCamel`](https://github.com/filecoin-project/js-filecoin-api-client/blob/master/src/lib/to-camel.js) utility.

e.g. `res.ProofInfo.PieceInclusionProof` should be accessed like `res.proofInfo.pieceInclusionProof`

hazmatzo replied:
Done!

alanshaw commented on API.md:
```diff
@@ -401,6 +401,55 @@ After first iteration:
+  "Message":"",
+  "ProposalCid":
+    {
+      "/":"zDPWYqFD8CNXu7Mo9qPSUANbTK2vi9vJBnvavF9S1pVGPHafVHpT"
```
This is an artifact of CID encoding in IPLD, to make it easier to work with we replace CID objects like this with the CID string. e.g.

https://github.com/filecoin-project/js-filecoin-api-client/blob/2c0e82fd8cfe7b959b1eb067a5d8e9960d2a16fe/src/cmd/actor/ls.js#L14

alanshaw commented:
It would be nice to get an e2e test for this (and other commands you're submitting). The current ones run against an already started daemon.

It would be even more rad if you were to tackle https://github.com/filecoin-project/js-filecoin-api-client/issues/69

hazmatzo commented:
> It would be nice to get an e2e test for this (and other commands you're submitting). The current ones run against an already started daemon.
> 
> It would be even more rad if you were to tackle #69

Hey @alanshaw, I'd be happy to tackle it but would probably need some additional context. I'm happy to start comments on the issue, but any chance you can chat?

alanshaw commented on src/cmd/client/query-storage-deal.js:
```diff
@@ -0,0 +1,24 @@
+
+    if (storageDeal.proofInfo) {
+      storageDeal.proofInfo = toCamel(storageDeal.proofInfo)
+    }
```
Need to fix the CID here too!

```suggestion
    }
    
    if (storageDeal.proofInfo && storageDeal.proofInfo.commitmentMessage) {
      storageDeal.proofInfo.commitmentMessage = storageDeal.proofInfo.commitmentMessage['/']
    }
```

hazmatzo replied:
Added it as part of the commit.

alanshaw commented:
> Hey @alanshaw, I'd be happy to tackle it but would probably need some additional context. I'm happy to start comments on the issue, but any chance you can chat?

I'm in #filecoin on freenode if you'd like to discuss!

hazmatzo commented:
Fixed up the linting and added the commitmentMessage in the new format. That had been an outstanding question in my mind.

I'm on #filecoin as `hazmatzo`. Give me 10 minutes and I can probably chat.

hazmatzo commented:
Oh, I'm on #filecoin Slack. I haven't joined the IRC...

hazmatzo commented:
@alanshaw Hey! Is there anything else? 

Relatedly, I looked into doing the e2e tests for any of these new endpoints, but to me it seems like I'd need the ability to spin up a new daemon for any of them?

mishmosh commented:
@alanshaw are you comfortable with merging this now that linting and other original comments have been fixed?

I believe @hazmatzo and @alanshaw chatted in more depth about e2e testing -- @hazmatzo could you document that in a separate issue?

alanshaw commented:
> I believe @hazmatzo and @alanshaw chatted in more depth about e2e testing -- @hazmatzo could you document that in a separate issue?

@mishmosh I made some progress towards this today and published https://www.npmjs.com/package/go-filecoin-dep. The AI is for me to document in https://github.com/filecoin-shipyard/js-filecoin-api-client/issues/69 the requirements for a module that is a utility to spawn a fresh filecoin node that we can use for e2e testing.

@hazmatzo this is great - thanks a lot ❤️
